javascript/javascript2/week3/FastetsPresser/index.js

```diff
@@ -0,0 +1,111 @@
+// Confetti function
+const canvas = document.querySelector("#my-confetti");
+const jsConfetti = new JSConfetti();
+
+let timerStarter;
+let timerStoped = true;
+let clInterval;
+let clInterval2;
+let s = 0;
+let l = 0;
+let userInput = document.querySelector("input");
+
+// Coundown before game started
+function countDownBeforeStart() {
+  if (userInput.value === " ".trim()) {
+    alert("Put valid number");
+  } else if (timerStoped === true) {
+    document.querySelector(".start").disabled = true;
+    s = 0;
+    document.querySelector(".sResult p").innerText = `${s}`;
+    l = 0;
+    document.querySelector(".lResult p").innerText = `${l}`;
+    document.querySelector("h3").innerHTML = ` `;
+    let secs = 4;
+    clInterval = setInterval(tick, 1000);
+    function tick() {
+      secs--;
+      document.querySelector("h2").innerHTML = secs;
+      if (secs === 0) {
+        clearInterval(clInterval);
+        document.querySelector("h2").innerHTML = `GO!`;
+        timer();
+      }
+    }
+  }
+}
+
+// Ingame countdown
+function timer() {
+  let roundTime = userInput.value;
+  clInterval2 = setInterval(tick, 1000);
+  function tick() {
+    roundTime--;
+    document.querySelector("h3").innerHTML = roundTime;
+    if (roundTime === 0) {
+      clearInterval(clInterval2);
+      document.querySelector("h3").innerHTML = `Game is over!`;
+    }
+  }
+  timerStoped = false;
+  s = 0;
+  l = 0;
+  document.querySelector(".lResult p").innerText = `${l}`;
+  document.querySelector(".sResult p").innerText = `${s}`;
+  window.addEventListener("keyup", count);
+  timerStarter = setTimeout(() => {
```

Comment:
Is it necessary for this setTimeout function to be assigned to a variable?

Reply:
Yes, and it's boolean. In some functions I'm checking it if it stops or not